packages/ckeditor5-ui/src/colorpicker/colorpickerview.ts

```diff
@@ -0,0 +1,136 @@
+/**
+ * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module ui/colorpicker/colorpickerview
+ */
+
+import { type Locale, global } from '@ckeditor/ckeditor5-utils';
+import { debounce, type DebouncedFunc } from 'lodash-es';
+import View from '../view';
+import type InputTextView from '../inputtext/inputtextview';
+import LabeledFieldView from '../labeledfield/labeledfieldview';
+import { createLabeledInputText } from '../labeledfield/utils';
+
+import 'vanilla-colorful/hex-color-picker.js';
+import '../../theme/components/colorpicker/colorpicker.css';
+
+export default class ColorPickerView extends View {
+	/**
+	 * Color picker component.
+	 */
+	declare public picker: ColorPickerType;
+
+	/**
+	 * Input to defining custom colors in HEX.
+	 */
+	declare public input: LabeledFieldView<InputTextView>;
+
+	/**
+	 * Current color state in color picker.
+	 */
+	declare public color: string;
+
+	/**
+	* Debounced event method. The `pickerEvent()` method is called the specified `waitingTime` after `debouncedPickerEvent()` is called,
+	* unless a new action happens in the meantime.
+	*/
+	declare private _debouncePickerEvent: DebouncedFunc< ( arg: string ) => void >;
+
+	/**
+	* Debounced event method. The `inputEvent()` method is called the specified `waitingTime` after `debouncedInputEvent()` is called,
+	* unless a new action happens in the meantime.
+	*/
+	declare private _debounceInputEvent: DebouncedFunc< ( arg: string ) => void >;
+
+	constructor( locale: Locale | undefined ) {
+		super( locale );
+
+		this.set( 'color', '' );
+
+		this.input = this._createInput();
+
+		const children = this.createCollection();
+		children.add( this.input );
+
+		this.setTemplate( {
+			tag: 'div',
+			attributes: {
+				class: [ 'ck', 'ck-color-picker' ]
+			},
+			children
+		} );
+
+		const waitingTime = 150;
+
+		this._debouncePickerEvent = debounce( ( color: string ) => {
+			this.fire( 'change', { value: color } );
+			this.color = color;
+		}, waitingTime );
+
+		this._debounceInputEvent = debounce( ( color: string ) => {
+			this.fire( 'change', { value: color } );
+			this.setColor( color );
+		}, waitingTime );
+	}
+
+	// Sets color in the color picker.
+	public setColor( color: string | undefined ): void {
```

Comment:
It would be nicer to turn it into a observable property (string) called `color`.

Then out of the box we can nicely observe it for state changes, etc.